app/services/post_alerter.rb

```diff
@@ -611,14 +613,22 @@ def notify_post_users(post, notified, include_category_watchers: true, include_t
 
     already_seen_users = TopicUser.where(topic_id: post.topic.id).where("highest_seen_post_number >= ?", post.post_number).pluck(:user_id)
 
-    notify.pluck(:id).each do |user_id|
-      notification_type = already_seen_users.include?(user_id) ? Notification.types[:edited] : Notification.types[:posted]
-      user = User.find_by(id: user_id)
+    each_user_in_batches(notify) do |user|
+      notification_type = already_seen_users.include?(user.id) ? Notification.types[:edited] : Notification.types[:posted]
```

Comment:
this is not new ... but `already_seen_users` should be a `Set` so include is fast and we are not constantly walking it.

Reply:
Also it should probably be called `already_seen_user_ids` since it's only holding user ids.